src/main/java/org/ow2/proactive/catalog/dto/BucketGrantMetadata.java

```diff
@@ -25,7 +25,10 @@
  */
 package org.ow2.proactive.catalog.dto;
 
+import java.util.Deque;
```

Comment:
Why are you using `Deque` why not `List`?

Reply:
We want a LIFO order. A list will provide the reverse.

Reply:
Fair enough. LGTM